### PR TITLE
Enable adding extra k8s resources to helm chart via values

### PR DESCRIPTION
**Pull request by jamesross3**
#### What this PR does
Originally here: https://github.com/grafana/mimir/pull/3956
Created a new PR after correcting local git config so that the license/cla check passes.

#### Which issue(s) this PR fixes or relates to

Fixes #<issue number>

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 * [CHANGE] Store-gateway: Remove experimental `-blocks-storage.bucket-store.max-concurrent-reject-over-limit` flag. #3706
 * [CHANGE] Ingester: If shipping is enabled block retention will now be relative to the upload time to cloud storage. If shipping is disabled block retention will be relative to the creation time of the block instead of the mintime of the last block created. #3816
 * [CHANGE] Query-frontend: Deprecated CLI flag `-query-frontend.align-querier-with-step` has been removed. #3982
+* [FEATURE] Helm Chart: Enable users to specify additional Kubernetes resource manifests using the `extraObjects` variable.
```
Comment (dimitarvdimitrov) [inline CHANGELOG.md:11]: Can you add the PR number in the entry, please?
```suggestion
* [FEATURE] Helm Chart: Enable users to specify additional Kubernetes resource manifests using the `extraObjects` variable. #3985
```

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM, with a small comment about the changelog entry. 

Thank you for the contribution :)

**Comment by 56quarters** [comment]
The CHANGELOG has just been cut to prepare for the next Mimir release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG document. Thanks!

**Comment by jamesross3** [comment]
thanks @dimitarvdimitrov! updated

**Comment by dimitarvdimitrov** [comment]
hey, I'm afraid this PR now has unintended changes. Maybe these got in during the rebase

Screenshot 2023-01-24 at 11 09 21

I also see a conflict with the changelog because I think the 2.6 Mimir release was cut. Could you look into that?

**Comment by jamesross3** [comment]
sorry @dimitarvdimitrov but replacing this again since I've messed up the commit history - see https://github.com/grafana/mimir/pull/4102 instead

**Pull request closed without merging**